src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,418 @@
+                builder.setMetrics(collectedMetrics);
+                builders.add(builder);
+            }
+            return builders;
+        } catch (final IOException ex) {
+            // CHECKSTYLE.OFF: IllegalInstantiation - Approved for byte[] to String
+            throw new ParsingException("Error parsing collectd json", data, ex);
+            // CHECKSTYLE.ON: IllegalInstantiation
+        }
+    }
+
+    private String computeMetricName(
+            final String plugin,
+            final String pluginInstance,
+            final String type,
+            final String typeInstance,
+            final String dsName) {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(plugin);
+        if (pluginInstance != null && pluginInstance.length() > 0) {
+            builder.append("/");
+            builder.append(pluginInstance);
+        }
+        builder.append("/");
+        builder.append(type);
+        if (typeInstance != null && typeInstance.length() > 0) {
+            builder.append("/");
+            builder.append(typeInstance);
+        }
+        if (dsName != null && dsName.length() > 0 && !dsName.equals("value")) {
+            builder.append("/");
+            builder.append(dsName);
+        }
+        return builder.toString();
+    }
+
+    private MetricType mapDsType(final String type) {
+        switch (type) {
+            case "gauge":
+                return MetricType.GAUGE;
+            case "absolute":
+                // This is an odd type.  It is a counter that is reset on read and divided by the last time.
+                return MetricType.COUNTER;
+            case "counter":
+                return MetricType.COUNTER;
+            case "derive":
+                return MetricType.COUNTER;
+            default:
+                return MetricType.GAUGE;
+        }
+    }
+
+    private static Metric mergeMetrics(final Map.Entry<String, Collection<Metric>> entries) {
+        final Collection<Metric> metrics = entries.getValue();
+        if (metrics.isEmpty()) {
+            throw new IllegalArgumentException("entries must not be empty");
+        }
+        final Metric firstMetric = metrics.iterator().next();
+        if (metrics.size() == 1) {
+            return firstMetric;
+        } else {
+            final List<Quantity> quantities = Lists.newArrayList();
+            for (Metric metric : metrics) {
+                quantities.addAll(metric.getValues());
+            }
+            return new DefaultMetric.Builder()
+                    .setType(firstMetric.getType())
+                    .setValues(quantities)
+                    .build();
+        }
+    }
+
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.createInstance();
+
+    static {
+        OBJECT_MAPPER.configure(JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS, true);
+        OBJECT_MAPPER.registerModule(new AfterburnerModule());
```

Comment:
I thought it could mess things up.  I don't really want to add it as a dependency everywhere either.